x/asset/client/cli/tx.go

```diff
@@ -160,27 +512,618 @@ func txUpdatePair() *cobra.Command {
+		if err != nil {
+			return txf, nil, err
+		}
+		newStabilityFee, err := sdk.NewDecFromStr(stabilityFee[i])
+		if err != nil {
+			return txf, nil, err
+		}
+		newClosingFee, err := sdk.NewDecFromStr(closingFee[i])
+		if err != nil {
+			return txf, nil, err
+		}
+		newLiquidationPenalty, err := sdk.NewDecFromStr(liquidationPenalty[i])
+		if err != nil {
+			return txf, nil, err
+		}
+		newDrawDownFee, err := sdk.NewDecFromStr(drawDownFee[i])
+		if err != nil {
+			return txf, nil, err
+		}
+		newMinCr, err := sdk.NewDecFromStr(minCr[i])
+		if err != nil {
+			return txf, nil, err
+		}
+		newIsVaultActive := ParseBoolFromString(isVaultActive[i])
+		if err != nil {
+			return txf, nil, err
+		}
+		debtCeiling, ok := sdk.NewIntFromString(debtCieling[i])
+		if !ok {
+			return txf, nil, types.ErrorInvalidDebtCeiling
+		}
+		newDebtFloor, ok := sdk.NewIntFromString(debtFloor[i])
+		if !ok {
+			return txf, nil, types.ErrorInvalidDebtFloor
+		}
+		newIsPsmPair := ParseBoolFromString(isPsmPair[i])
+		newAssetOutOraclePrice := ParseBoolFromString(assetOutOraclePrice[i])
+		pairs = append(pairs, types.ExtendedPairVault{
+			AppMappingId:        appMappingId,
+			PairId:              pairId[i],
+			LiquidationRatio:    newLiquidationRatio,
+			StabilityFee:        newStabilityFee,
+			ClosingFee:          newClosingFee,
+			LiquidationPenalty:  newLiquidationPenalty,
+			DrawDownFee:         newDrawDownFee,
+			IsVaultActive:       newIsVaultActive,
+			DebtCeiling:         debtCeiling,
+			DebtFloor:           newDebtFloor,
+			IsPsmPair:           newIsPsmPair,
+			MinCr:               newMinCr,
+			PairName:            pairName[i],
+			AssetOutOraclePrice: newAssetOutOraclePrice,
+			AssetOutPrice:       assetOutPrice[i],
+			MinUsdValueLeft:     minUsdValueLeft[i],
+		})
+	}
+
+	deposit, err := sdk.ParseCoinsNormalized(extPairVault.Deposit)
+	if err != nil {
+		return txf, nil, err
+	}
+
+	content := types.NewAddExtendedPairsVaultProposa(extPairVault.Title, extPairVault.Description, pairs)
+
+	msg, err := govtypes.NewMsgSubmitProposal(content, deposit, from)
+	if err != nil {
+		return txf, nil, err
+	}
+
+	if err = msg.ValidateBasic(); err != nil {
+		return txf, nil, err
+	}
+	return txf, msg, nil
+}
+
+func NewCreateAssetMappingMsg(clientCtx client.Context, txf tx.Factory, fs *flag.FlagSet) (tx.Factory, sdk.Msg, error) {
+	assetMapping, err := parseAssetMappingFlags(fs)
+	if err != nil {
+		return txf, nil, fmt.Errorf("failed to parse assetMapping: %w", err)
+	}
+
+	appId, err := strconv.ParseUint(assetMapping.AppId, 10, 64)
+	if err != nil {
+		return txf, nil, err
+	}
+
+	assetId, err := ParseUint64SliceFromString(assetMapping.AssetId, ",")
+	if err != nil {
+		return txf, nil, err
+	}
+	genesisSupply, err := ParseStringFromString(assetMapping.GenesisSupply, ",")
+	if err != nil {
+		return txf, nil, err
+	}
+	isGovToken, err := ParseStringFromString(assetMapping.IsGovToken, ",")
```

Comment:
any reason we are not using string instead @cgsingh33 @rajrohit10

Reply:
same as above for adding proposal for multiple asset mapping.